### PR TITLE
Fix for devices that does not support fp64

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

File: onedal/datatypes/_data_conversion.py [hunk 1]
```diff
@@ -51,18 +52,21 @@ def _convert_to_supported_impl(policy, *data):
...
         if not device.has_aspect_fp64:
-            return (convert_or_pass(x) for x in data)
+            return [convert_or_pass(x) for x in data]
         else:
-            return data
+            return [x for x in data]
 else:
     def _convert_to_supported_impl(policy, *data):
-        return data
+        return [x for x in data]
```
Comment (Alexsandruss) [inline onedal/datatypes/_data_conversion.py:69]: Why list instead of tuple is returned now?

Comment (KulikovNikita) [reply onedal/datatypes/_data_conversion.py]: Fixed

**Comment by KulikovNikita** [comment]
http://intel-ci.intel.com/edb80e8e-4235-f1df-af9d-a4bf010d0e2e

File: onedal/datatypes/_data_conversion.py [hunk 2]
```diff
@@ -29,10 +30,14 @@ def convert_one_to_table(arg):
     return _backend.to_table(arg)
 
 
-def to_table(*args):
+def _apply_and_pass(func, *args):
     if len(args) == 1:
-        return convert_one_to_table(args[0])
-    return (convert_one_to_table(item) for item in args)
+        return func(args[0])
+    return (func(item) for item in args)
```
Comment (Alexsandruss) [inline onedal/datatypes/_data_conversion.py:36]:
```suggestion
    return tuple(map(func, args))
```

File: onedal/primitives/kernel_functions.py [hunk 3]
```diff
@@ -131,7 +132,7 @@ def sigmoid_kernel(X, Y=None, gamma=1.0, coef0=0.0, queue=None):
     kernel_matrix : ndarray of shape (n_samples_X, n_samples_Y)
     """
 
-    X, Y, fptype = _check_inputs(X, Y)
-    return _compute_kernel({'fptype': fptype, 'method': 'dense',
+    X, Y = _check_inputs(X, Y)
+    return _compute_kernel({'method': 'dense',
```
Comment (Alexsandruss) [inline onedal/primitives/kernel_functions.py:136]: Why type is removed?

File: onedal/linear_model/tests/test_linear_regression.py [hunk 4]
```diff
@@ -71,15 +72,15 @@ def test_full_results(queue):
         model = LinearRegression(fit_intercept=True)
         model.fit(X, y, queue=queue)
 
-        assert_allclose(coef, model.coef_.T)
-        assert_allclose(intp, model.intercept_)
+        assert_allclose(coef, model.coef_.T, rtol=2e-3)
+        assert_allclose(intp, model.intercept_, rtol=1e-3)
```
Comment (Alexsandruss) [inline onedal/linear_model/tests/test_linear_regression.py:76]: Is not rtol too big? What is justification of that?

File: onedal/primitives/kernel_functions.py (hunk 3 above)
Comment (KulikovNikita) [reply onedal/primitives/kernel_functions.py:136]: `fptype` is not deleted but moved to this [location](https://github.com/KulikovNikita/scikit-learn-intelex/blob/174f2a1422c675f7da66271a2045aa7b1a1e6171/onedal/primitives/kernel_functions.py#L36)

File: onedal/datatypes/_data_conversion.py (hunk 2 above)
Comment (KulikovNikita) [reply onedal/datatypes/_data_conversion.py]: Applied

File: onedal/linear_model/tests/test_linear_regression.py (hunk 4 above)
Comment (KulikovNikita) [reply onedal/linear_model/tests/test_linear_regression.py]: Changed

**Comment by KulikovNikita** [comment]
http://intel-ci.intel.com/edb8238e-fa2e-f14c-a520-a4bf010d0e2e

**Comment by KulikovNikita** [comment]
http://intel-ci.intel.com/edbc06ee-3f88-f1b2-b3bc-a4bf010d0e2e

**Comment by KulikovNikita** [comment]
Master CI job for comparison: http://intel-ci.intel.com/edbc133b-89cb-f124-9878-a4bf010d0e2e

File: onedal/datatypes/_data_conversion.py [hunk 5]
```diff
@@ -43,32 +46,37 @@ def to_table(*args):
...
 
-def _convert_to_supported(policy, *data):
-    res = _convert_to_supported_impl(policy, *data)
 
-    if len(data) == 1:
-        return res[0]
-    else:
-        return res
+def _convert_to_supported(policy, *data):
+    return _convert_to_supported_impl(policy, *data)
```
Comment (Alexsandruss) [inline onedal/datatypes/_data_conversion.py:82]: Meaningless, rename _convert_to_supported_impl to _convert_to_supported and remove this

File: onedal/linear_model/linear_model.py [hunk 6]
```diff
@@ -48,11 +48,12 @@ def _get_policy(self, queue, *data):
...
     def _get_onedal_params(self, dtype=np.float32):
         intercept = 'intercept|' if self.fit_intercept else ''
-        return {
-            'fptype': 'float' if dtype is np.float32 else 'double',
+        res = {
+            'fptype': 'float' if dtype == np.float32 else 'double',
             'method': self.algorithm, 'intercept': self.fit_intercept,
             'result_option': (intercept + 'coefficients'),
         }
+        return res
```
Comment (Alexsandruss) [inline onedal/linear_model/linear_model.py:56]:
```suggestion
        return {
            'fptype': 'float' if dtype == np.float32 else 'double',
            'method': self.algorithm, 'intercept': self.fit_intercept,
            'result_option': (intercept + 'coefficients'),
        }
```

File: onedal/linear_model/tests/test_linear_regression.py [hunk 7]
```diff
@@ -53,76 +56,86 @@ def test_pickle(queue):
 
         assert isinstance(model2, model.__class__)
         result = model2.predict(X, queue=queue)
-        assert_array_equal(expected, result)
+
+        assert_allclose(expected, result, rtol=1e-5)
```
Comment (Alexsandruss) [inline onedal/linear_model/tests/test_linear_regression.py:60]: `rtol=1e-5` for original and pickled model is confusing. Why it should have this value?

Comment (KulikovNikita) [reply onedal/linear_model/tests/test_linear_regression.py]: Done

File: onedal/linear_model/linear_model.py (hunk 6 above)
Comment (KulikovNikita) [reply onedal/linear_model/linear_model.py]: Done

File: onedal/datatypes/_data_conversion.py (hunk 5 above)
Comment (KulikovNikita) [reply onedal/datatypes/_data_conversion.py]: Done

**Comment by KulikovNikita** [comment]
http://intel-ci.intel.com/edbccedf-ec3f-f1b7-9e14-a4bf010d0e2e

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**